lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -170,6 +183,30 @@ spec = do
             expectResponseCode @IO HTTP.status405 r
             expectErrorMessage errMsg405 r
 
+    it "STAKE_POOLS_LIST_04 - Discovers new pools when they are registered" $ \ctx -> do
```

Comment:
@rvl I added a rather extensive integration test. The setup behind `registerStakePool` is a bit complex because we have to do everything with `jcli` and use accounts to pay for the registration certificate! I got to the end of it and.. nicely caught a small bug in the "zipping logic" from the `listStakePools` endpoint (cf my last commit). 
Please have another round of review / fixes and let's get this merged! 

Some outstanding points remaining after that:

- Maybe do some caching? I expect this endpoint to be quite heavily hammered by clients like Daedalus. The registry won't change very often. Renewing a cache every hour or so should also mitigate the concern. 
- Currently the handler for `joinStakePool` relies on `listStakePool` to get a list of known stake pool... that's kinda lame because `listStakePool` is a very heavy operation and, all stake pools ids can be known by simply making a database call. So we should fix that.
- https://github.com/input-output-hk/cardano-wallet/issues/1158